src/contact/reducers/Reducer.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import * as ActionTypes from '../actions/ActionTypes';
+
+const INITIAL_STATE = {
+	details : null,
+	isEditable : false
+};
+
+export default function Reducer(state = INITIAL_STATE, action) {
+	switch (action.type) {
+		case ActionTypes.DISPLAY_INFO: {
+			return Object.assign({}, state, {
+				details: action.payload,
+				isEditable: action.payload.isEditable
+			});
+		}
+		case ActionTypes.ADD_EMAIL: {
+			const length = (Object.keys(state.details.otheremails).length)+1;
```

Comment:
How can you define same const length variable so many times? Whats the meaning of const then?